src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SyntaxEquivalentNode.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Interop
+{
+    public record struct SyntaxEquivalentNode<T>(T Node)
+        where T : SyntaxNode
+    {
+        public bool Equals(SyntaxEquivalentNode<T> other)
+        {
+            return SyntaxEquivalentComparer.Instance.Equals(Node, other.Node);
+        }
+
+        public override int GetHashCode() => SyntaxEquivalentComparer.Instance.GetHashCode(Node);
```

Comment:
SyntaxEquivalentComparer.Instance.GetHashCode(Node) throws UnreachableException. I feel like we should either implement that method or throw here.